src/ring_theory/polynomial/chebyshev/defs.lean

```diff
@@ -112,4 +126,167 @@ begin
+          + (2 * X * chebyshev₁ R (n + 2) - chebyshev₁ R (n + 1))
+                : by ring
+  ... = X * chebyshev₂ R (n + 2) + chebyshev₁ R (n + 2 + 1)
+                : by simp only [chebyshev₂_add_two, chebyshev₁_add_two]
+end
+
+lemma chebyshev₁_eq_chebyshev₂_sub_X_mul_chebyshev₂ (n : ℕ) :
+chebyshev₁ R (n+1) = chebyshev₂ R (n+1) - X * chebyshev₂ R n :=
+by rw [chebyshev₂_eq_X_mul_chebyshev₂_add_chebyshev₁, add_comm (X * chebyshev₂ R n), add_sub_cancel]
+
+
+lemma chebyshev₁_eq_X_mul_chebyshev₁_sub_pol_chebyshev₂ :
+∀ (n : ℕ), chebyshev₁ R (n+2) = X * chebyshev₁ R (n+1) - (1 - X ^ 2) * chebyshev₂ R n
+|0        := by {simp only [chebyshev₁_one, chebyshev₁_two, chebyshev₂_zero], ring}
+|1        := by {simp only [chebyshev₁_add_two, chebyshev₁_zero, chebyshev₁_add_two,
+                              chebyshev₂_one, chebyshev₁_one], ring }
+|(n + 2)  := begin
+calc chebyshev₁ R (n + 2 + 2) = 2 * X * chebyshev₁ R (n + 2 + 1) - chebyshev₁ R (n + 2)
+                                : chebyshev₁_add_two _ _
+... = 2 * X * (X * chebyshev₁ R (n + 2) - (1 - X ^ 2) * chebyshev₂ R (n + 1))
+                                        - (X * chebyshev₁ R (n + 1) - (1 - X ^ 2) * chebyshev₂ R n)
+                          : by simp only [chebyshev₁_eq_X_mul_chebyshev₁_sub_pol_chebyshev₂]
+... = X * (2 * X * chebyshev₁ R (n + 2) -  chebyshev₁ R (n + 1))
+                    - (1 - X ^ 2) * (2 * X * chebyshev₂ R (n + 1) - chebyshev₂ R n)
+                          : by ring
+... = X * chebyshev₁ R (n + 2 + 1) - (1 - X ^ 2) * chebyshev₂ R (n + 2)
+                          : by rw [chebyshev₁_add_two _ (n + 1), chebyshev₂_add_two],
+end
+
+lemma one_sub_X_pow_two_mul_chebyshev₂_eq_pol_in_chebyshev₁ (n : ℕ) :
+  (1 - X ^ 2) * chebyshev₂ R n = X * chebyshev₁ R (n + 1) - chebyshev₁ R (n + 2) :=
+by rw [chebyshev₁_eq_X_mul_chebyshev₁_sub_pol_chebyshev₂, ←sub_add, sub_self, zero_add]
+
+variables {R S}
+
+lemma map_chebyshev₂ (f : R →+* S) :
+  ∀ (n : ℕ), map f (chebyshev₂ R n) = chebyshev₂ S n
+| 0       := by simp only [chebyshev₂_zero, map_one]
+| 1       := begin simp only [chebyshev₂_one, map_X, map_mul, map_add, map_one],
+                   change map f (1+1) * X = 2 * X,
+                   simpa only [map_add, map_one] end
+| (n + 2) :=
+begin
+  simp only [chebyshev₂_add_two, map_mul, map_sub, map_X, bit0, map_add, map_one],
+  rw [map_chebyshev₂ (n + 1), map_chebyshev₂ n],
+end
+
+lemma chebyshev₁_derivative_eq_chebyshev₂ :
+∀ (n : ℕ), derivative (chebyshev₁ R (n + 1)) = (n + 1) * chebyshev₂ R n
+|0        := by simp only [chebyshev₁_one, chebyshev₂_zero, derivative_X, nat.cast_zero, zero_add,
+                           mul_one]
+|1        := by { simp only [chebyshev₁_two, chebyshev₂_one, derivative_sub, derivative_one,
+                              derivative_mul, derivative_X_pow, nat.cast_one,
+                              nat.cast_two],
+                    norm_num }
+|(n + 2)  := begin
+  calc derivative (chebyshev₁ R (n + 2 + 1))
```

Comment:
`calc` can be used directly in term mode. So you can remove the `begin` and `end`.